ddtrace/utils/version.py

```diff
@@ -0,0 +1,26 @@
+import typing
+
+import packaging.version
```

Comment:
I'm wondering whether we could do away with an extra dependency, considering that all we need is just a small feature that this package offers.

Would something like this be enough for our needs?

~~~ python
from itertools import takewhile


def parse_version(version):
    return (
        tuple(
            int(_)
            for _ in "".join(
                list(takewhile(lambda _: _.isdigit() or _ == ".", version))
            ).split(".")
        )
        + (0, 0, 0)
    )[:3]
~~~

Reply:
Just my opinion but it seems unnecessary to add in another dependency especially since all we need is this small feature that seems trivial to implement. Does `takewhile` provide a performance benefit compared to our current implementation?

Reply:
Considering that version strings are generally short, and that we probably don't have version parsing on a hot path, there shouldn't be any noticeable performance difference between the two implementations.

Reply:
In that case, I'm leaning towards the current approach on the basis that it's easier to read 😆

Reply:
No reason to make it a one-liner :). It could easily be split to make it more readable, e.g.,
~~~ python
# Keep only initial digits and dots
version_head = "".join(list(takewhile(lambda _: _.isdigit() or _ == ".", version)))
# Return (major, minor, patch)
return (tuple(int(_) for _ in version_head.split(".")) + (0, 0, 0))[:3]
~~~

Reply:
if it helps, `packaging` is already a dependency that exists/is available to us via `pkg_resources`, however, we are adding it as an explicit dependency in `setup.py` to be more clear and ensure specific features exist.

as far as performance improvements, `packaging.version.Version` is just a single regex with matching groups. This could will _most likely_ only be run on startup/import anyways so the impact should be negligible (although, I will admit I am not measuring it, so that is a biases/unproven statement about performance).

Reply:
> if it helps, `packaging` is already a dependency that exists/is available to us via `pkg_resources`

Ah, I didn't know that! In that case it makes sense to leverage it if it's already around. Like I said in my reply to @Yun-Kim, I do not have performance concerns here, just trying to avoid a potentially new dependency.